src/utils/testing.ts

```diff
@@ -3,6 +3,193 @@ import { SchematicTestRunner, UnitTestTree } from '@angular-devkit/schematics/te
 import { Tree } from '@angular-devkit/schematics';
 
 const collectionPath = join('./node_modules/@schematics/angular/collection.json');
+const commonJsFile = 
+`({
+    rules: [
+        { test: /.anything$/, loader: 'any-loader' }
+    ]
+})`
+
+const angular8TypescriptJsFile =
```

Comment:
CLI for Angular 7 [also has the `directTemplateLoading` flag set](https://github.com/angular/angular-cli/blob/v7.2.0/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/typescript.ts#L85) (was first added at v7.2.0), that's why I assumed #19 would also be fixed...

That being said, `createCommonWebpackConfigForAngular8` should be "7 or greater", and `6or7` should be `6` only.

Reply:
A tip about unit tests: as you know how the code works when writing them, you can be more specific on the mocked test data. Instead of using the whole `typescript.js` file, you could create a subset of it, with just the parts that the code and test interact with. An example with the current code:

**`angular6or7TypescriptJsFile`**
```
  const pluginOptions: AngularCompilerPluginOptions = {
    mainPath: useMain ? path.join(root, buildOptions.main) : undefined,
    ...i18nFileAndFormat,
    locale: buildOptions.i18nLocale,
    platform: buildOptions.platform === 'server' ? PLATFORM.Server : PLATFORM.Browser,
    missingTranslation: buildOptions.i18nMissingTranslation,
    sourceMap: buildOptions.sourceMap.scripts,
    additionalLazyModules,
    hostReplacementPaths,
    nameLazyFiles: buildOptions.namedChunks,
    forkTypeChecker: buildOptions.forkTypeChecker,
    contextElementDependencyConstructor: require('webpack/lib/dependencies/ContextElementDependency'),
    logger: wco.logger,
    ...options,
  };
```

**`angular8TypescriptJsFile`**
```
  const pluginOptions: AngularCompilerPluginOptions = {
    mainPath: useMain ? path.join(root, buildOptions.main) : undefined,
    ...i18nFileAndFormat,
    locale: buildOptions.i18nLocale,
    platform: buildOptions.platform === 'server' ? PLATFORM.Server : PLATFORM.Browser,
    missingTranslation: buildOptions.i18nMissingTranslation,
    sourceMap: buildOptions.sourceMap.scripts,
    additionalLazyModules,
    hostReplacementPaths,
    nameLazyFiles: buildOptions.namedChunks,
    forkTypeChecker: buildOptions.forkTypeChecker,
    contextElementDependencyConstructor: require('webpack/lib/dependencies/ContextElementDependency'),
    logger: wco.logger,
    directTemplateLoading: true,
    ...options,
  };
```

Reply:
Thanks for the tip!